### PR TITLE
DEVPROD-10458 fix update permissions route for repo projects

### PR DESCRIPTION
**Pull request by ablack12**
DEVPROD-10458 

### Description
I don't love this, but Mana doesn't work for repo permissions, so adding some special casing. I'm not sure of a better long-term solution -- seems like not something we can ask mana to handle since it involves really knowing how things work for on our end. Possibly we could remove the concept of view permission for repos and just assume view if you have view access for a project, but idk about that either. 

Possibly a true ideal solution would be making these pages public once private vars are moved over.

### Testing
Added a unit test.

### CODE REVIEW

File: rest/route/user.go [hunk 1]
```diff
@@ -210,6 +210,29 @@ func (h *userPermissionsPostHandler) Run(ctx context.Context) gimlet.Responder {
 		return gimlet.NewTextInternalErrorResponse(err.Error())
 	}
 
+	// This is unfortunately very special-casey, but if the user has been granted permission to view/edit
+	// project settings, they also need to be given view access for the repo project, if applicable.
```
Comment (malikchaya2) [inline rest/route/user.go:214]: Given this change, I think we should also update the text in the UI where admins are added 

> Admins for this branch will be able to edit branch settings. All admins will have access to create new projects on Evergreen **and view the repo if the project is attached**.

Comment (malikchaya2) [inline rest/route/user.go:214]: By `if applicable` do you mean `if the project is attached`? Can we update this comment to be more specific about when someone is given view access?

**Review by malikchaya2 — APPROVED** [review]

File: rest/route/user.go [hunk 2]
```diff
@@ -210,6 +210,29 @@ func (h *userPermissionsPostHandler) Run(ctx context.Context) gimlet.Responder {
...
+		h.permissions.Permissions[evergreen.PermissionProjectSettings] >= evergreen.ProjectSettingsView.Value {
+		repoProjectsUpdated := map[string]bool{}
+		pRefs, err := serviceModel.FindMergedEnabledProjectRefsByIds(h.permissions.Resources...)
+		if err != nil {
+			return gimlet.NewTextInternalErrorResponse(fmt.Sprintf(
+				"problem checking for repos: %s", err))
+		}
+
+		for _, pRef := range pRefs {
+			fmt.Println("repo ref ", pRef.RepoRefId)
```
[the remark this answers is not in the document]
Comment (ablack12) [reply rest/route/user.go:225]: imagine if i said yes haha

File: rest/route/user.go (hunk 1 above)
Comment (ablack12) [reply rest/route/user.go]: we actually already have "and view repo settings" for attached projects:
image

Comment (ablack12) [reply rest/route/user.go]: sure

**Pull request merged**